src/types/message.rs

```diff
@@ -1095,6 +1114,93 @@ impl Message {
         // /[a-zA-Z0-9_]{5,32}/ and chat/message ids are integers.
         Some(reqwest::Url::parse(&url).unwrap())
     }
+
+    /// Produces a direct link to a comment on this post.
+    ///
+    /// Note that for private channels the link will only be accessible for
+    /// channel members.
+    ///
+    /// Returns `None` for private chats (i.e.: DMs) and private groups (not
+    /// supergroups).
+    pub fn comment_url(&self, comment_id: i32) -> Option<Url> {
+        Self::comment_url_of(self.chat.id, self.chat.username(), self.id, comment_id)
+    }
+
+    /// Produces a direct link to a comment on a post.
+    ///
+    /// If you have a `Message` object of the channel post, use [`comment_url`]
+    /// instead. This function should only be used if you have limited
+    /// information about the message (channel id, username of the channel,
+    /// if any, post id and comment id).
+    ///
+    /// Note that for private channels the link will only be accessible for
+    /// channel members.
+    ///
+    /// Returns `None` for private chats (i.e.: DMs) and private groups (not
+    /// supergroups).
+    ///
+    /// [`comment_url`]: Message::comment_url
+    pub fn comment_url_of(
+        channel_id: ChatId,
+        channel_username: Option<&str>,
+        post_id: i32,
+        comment_id: i32,
+    ) -> Option<Url> {
+        Self::url_of(channel_id, channel_username, post_id).map(|mut url| {
+            url.set_query(Some(&format!("comment={comment_id}")));
+            url
+        })
+    }
+
+    /// Produces a direct link to this message in a given thread.
+    ///
+    /// "Thread" is a group of messages that reply to each other in a tree-like
+    /// structure. `thread_starter_msg_id` is the id of the first message in
+    /// the thread, the root of the tree.
+    ///
+    /// Note that for private groups the link will only be accessible for group
+    /// members.
+    ///
+    /// Returns `None` for private chats (i.e.: DMs) and private groups (not
+    /// supergroups).
+    pub fn url_in_thread(&self, thread_starter_msg_id: i32) -> Option<Url> {
```

Comment:
imho, thread_starter_msg_id may be optional and if it's none, then you can retrieve thread from reply_to of self

Reply:
you can't if this is a second level reply

Reply:
yep, And it's reason to make that parameter optional

Reply:
what?...

if this message is a second level reply then there is literally no way to make an url without having some kind of a database